File: finstack-py/tests/golden/runners/integration_credit.py
"""Domain runner for credit integration golden fixtures."""

from __future__ import annotations

from tests.golden.runners.integration_common import run_credit_integration
from tests.golden.schema import GoldenFixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Require executable credit integration inputs before running."""
    return run_credit_integration(fixture)

File: finstack/valuations/tests/golden/runners/integration_credit.rs
//! Domain runner for credit calibrate-then-price integration goldens.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Credit integration golden runner.
pub struct IntegrationCreditRunner;

impl DomainRunner for IntegrationCreditRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::integration_common::run_credit_integration(fixture)
    }
}
